units/provisioning_setup_host.go

```diff
@@ -105,7 +113,6 @@ func (j *setupHostJob) Run(ctx context.Context) {
 	if j.env == nil {
 		j.env = evergreen.GetEnvironment()
 	}
-	defer j.tryRequeue(ctx)
```

Comment:
If we remove `tryRequeue()`, we're instead relying on amboy re-enqueueing a job when an attempt fails. But there will be a problem with multiple provisioning jobs running for a single host since we don't have scopes and job uniqueness is determined by provisioning attempt, which is incremented at the beginning of `provisionHost()`

I would move `IncProvisionAttempts()` and put it in a defer within `setupHost()` to ensure that all SSHing/provisioning/DB updating is done by the time the provisioning attempt is incremented.

Reply:
Could we also solve this by putting a timeout on the context that is less than the amount of time between job recreation (assuming we check that context in enough places)?

Reply:
what if we added the current provision number to the query on inc findAndModify, so that it fails (and we abort the job if another job has started, so that the attempt number works like a lock?

Reply:
I'm fine with making IncProvisionAttempts the lock.